File: backend/src/appointment/entities/appointment.entity.ts
import * as mongose from 'mongoose'

export const AppointmentSchema = new mongose.Schema({
    title: { type: String, required: true},
    body: { type: String, required: true},
    date: { type: String, required: true},
    time: { type: String, required: true},
    author: { type: String, required: true},

}, {timestamps:true})

export interface Appointment {
    id: string;
    title: string;
    body: string;
    date: string;
    time: string;
    author: string;

}

File: frontend/lib/domain/appointment/local/appointment_entity.dart
class AppointmentEntity {
  final String id;
  final String title;
  final String body;
  final String date;
  final String time;
  final String author;

  AppointmentEntity({
    required this.id,
    required this.title,
    required this.body,
    required this.date,
    required this.time,
    required this.author,
  });

  factory AppointmentEntity.fromJson(Map<String, dynamic> json) {
    return AppointmentEntity(
      id: json['id'] as String,
      title: json['title'] as String,
      body: json['body'] as String,
      date: json['date'] as String,
      time: json['time'] as String,
      author: json['author'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'body': body,
      'date': date,
      'time': time,
      'author': author,
    };
  }
}
